From 99691358552207f1d7a2f6600dfe1a6462f6ffd3 Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Thu, 11 Aug 2022 02:14:22 +0000
Subject: [PATCH] Revert "HSD #15010938416: arm: dts: soc64: changing DDR
 aliasing addresses"

This reverts commit 9cee17313ab30b71dc53a98035c2ce3c3e883de0.

with this patch the pcie feature on stratix 10 will not work well
and the ethernet card plug into pcie slot will continuely print
below error:

e1000e 0000:01:00.0 eth0: Reset adapter unexpectedly
e1000e 0000:01:00.0 eth0: Hardware Error

revert this patch as a wrokaround until a best fix can be made.

Upstream-Status: Pending

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 arch/arm/dts/socfpga_agilex_socdk-u-boot.dtsi | 26 +++----------------
 arch/arm/dts/socfpga_stratix10_socdk.dts      | 26 +++----------------
 .../socfpga_stratix10_socdk_nand-u-boot.dtsi  | 26 ++-----------------
 3 files changed, 8 insertions(+), 70 deletions(-)

diff --git a/arch/arm/dts/socfpga_agilex_socdk-u-boot.dtsi b/arch/arm/dts/socfpga_agilex_socdk-u-boot.dtsi
index 8d50271b4962..6efe2f0952a3 100644
--- a/arch/arm/dts/socfpga_agilex_socdk-u-boot.dtsi
+++ b/arch/arm/dts/socfpga_agilex_socdk-u-boot.dtsi
@@ -2,7 +2,7 @@
 /*
  * U-Boot additions
  *
- * Copyright (C) 2019-2022 Intel Corporation <www.intel.com>
+ * Copyright (C) 2019 Intel Corporation <www.intel.com>
  */
 
 #include "socfpga_agilex-u-boot.dtsi"
@@ -28,29 +28,9 @@
 	};
 
 	memory {
-		/*
-		 * Recommended Aliasing addresses
-		 *
-		 * 16GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 3 0x80000000>;
-		 *
-		 * 8GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 1 0x80000000>;
-		 *
-		 * 4GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 0 0x80000000>;
-		 *
-		 * 2GB
-		 *     <0 0x00000000 0 0x80000000>;
-		 *
-		 * Note: Need to set CONFIG_NR_DRAM_BANKS=1 for 2GB in defconfig
-		 *       Default CONFIG_NR_DRAM_BANKS=2 is used for other DDR size
-		 */
+		/* 8GB */
 		reg = <0 0x00000000 0 0x80000000>,
-		      <0x10 0x80000000 1 0x80000000>;
+		      <2 0x80000000 1 0x80000000>;
 	};
 };
 
diff --git a/arch/arm/dts/socfpga_stratix10_socdk.dts b/arch/arm/dts/socfpga_stratix10_socdk.dts
index 237461f943db..6bffd0703242 100755
--- a/arch/arm/dts/socfpga_stratix10_socdk.dts
+++ b/arch/arm/dts/socfpga_stratix10_socdk.dts
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright (C) 2018-2022 Intel Corporation
+ * Copyright (C) 2018 Intel Corporation
  */
 
 #include "socfpga_stratix10.dtsi"
@@ -36,29 +36,9 @@
 		#address-cells = <2>;
 		#size-cells = <2>;
 		device_type = "memory";
-		/*
-		 * Recommended Aliasing addresses
-		 *
-		 * 16GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 3 0x80000000>;
-		 *
-		 * 8GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 1 0x80000000>;
-		 *
-		 * 4GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 0 0x80000000>;
-		 *
-		 * 2GB
-		 *     <0 0x00000000 0 0x80000000>;
-		 *
-		 * Note: Need to set CONFIG_NR_DRAM_BANKS=1 for 2GB in defconfig
-		 *       Default CONFIG_NR_DRAM_BANKS=2 is used for other DDR size
-		 */
+		/* 4GB */
 		reg = <0 0x00000000 0 0x80000000>,
-		      <0x10 0x80000000 0 0x80000000>;
+		      <1 0x80000000 0 0x80000000>;
 		u-boot,dm-pre-reloc;
 	};
 };
diff --git a/arch/arm/dts/socfpga_stratix10_socdk_nand-u-boot.dtsi b/arch/arm/dts/socfpga_stratix10_socdk_nand-u-boot.dtsi
index 7797238e7c5c..ae859ab42ec4 100644
--- a/arch/arm/dts/socfpga_stratix10_socdk_nand-u-boot.dtsi
+++ b/arch/arm/dts/socfpga_stratix10_socdk_nand-u-boot.dtsi
@@ -1,8 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
  * Copyright Altera Corporation (C) 2020. All rights reserved.
- *
- * Copyright (C) 2022 Intel Corporation
  */
 
 #include "socfpga_stratix10_socdk-u-boot.dtsi"
@@ -12,29 +10,9 @@
 		#address-cells = <2>;
 		#size-cells = <2>;
 		device_type = "memory";
-		/*
-		 * Recommended Aliasing addresses
-		 *
-		 * 16GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 3 0x80000000>;
-		 *
-		 * 8GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 1 0x80000000>;
-		 *
-		 * 4GB
-		 *     <0 0x00000000 0 0x80000000>,
-		 *     <0x10 0x80000000 0 0x80000000>;
-		 *
-		 * 2GB
-		 *     <0 0x00000000 0 0x80000000>;
-		 *
-		 * Note: Need to set CONFIG_NR_DRAM_BANKS=1 for 2GB in defconfig
-		 *       Default CONFIG_NR_DRAM_BANKS=2 is used for other DDR size
-		 */
+		/* 4GB */
 		reg = <0 0x00000000 0 0x80000000>,
-		      <0x10 0x80000000 0 0x80000000>;
+		      <1 0x80000000 0 0x80000000>;
 		u-boot,dm-pre-reloc;
 	};
 
-- 
2.32.0

